icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6741,3 +6742,17 @@ static void TestBug20149() {
         log_err("ERROR: in uloc_forLanguageTag %s return %s\n", myErrorName(status), locale);
     }
 }
+
+// Test case for ICU-20575
+// This test checks if the environment variable LANG is set, 
+// and if so ensures that C/C.UTF-8 cause ICU's default locale to be en_US_POSIX.
```

Comment:
nit: I'd probably suggest making it a bit more explicit:
"ensures that C/C.UTF-8" --> "ensures that both "C" and "C.UTF-8" cause"